File: Serene/Serene.Script/Imports/ServiceContracts/Administration.RoleRow.cs

namespace Serene.Administration
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class RoleRow
    {
        public Int32? RoleId { get; set; }
        public String RoleName { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string RoleId = "RoleId";
            [InlineConstant] public const string RoleName = "RoleName";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Imports/ServiceTypings/Administration.RoleRow.ts
namespace Serene.Administration
{
    export interface RoleRow
    {
        RoleId: number;
        RoleName: string;
    }

    export namespace RoleRow
    {
        export const IdProperty = "RoleId";
        export const NameProperty = "RoleName";
        export const LocalTextPrefix = "Administration.Role";
        export const LookupKey = "Administration.Role";

        export namespace Fields
        {
            export declare const RoleId: "RoleId";
            export declare const RoleName: "RoleName";
        }

        ["RoleId","RoleName"].forEach(x => (<any>Fields)[x] = x);
    }
}

